compiler/rustc_codegen_llvm/src/builder.rs

```diff
@@ -155,8 +155,14 @@ impl<'a, 'll, 'tcx> BuilderMethods<'a, 'tcx> for Builder<'a, 'll, 'tcx> {
 
     fn append_block(cx: &'a CodegenCx<'ll, 'tcx>, llfn: &'ll Value, name: &str) -> &'ll BasicBlock {
         unsafe {
-            let name = SmallCStr::new(name);
-            llvm::LLVMAppendBasicBlockInContext(cx.llcx, llfn, name.as_ptr())
+            let c_str_name;
+            let name_ptr = if cx.tcx.sess.fewer_names() {
```

Comment:
Isn't `fewer_names` off by default, so need to revert this condition to actually test perf?

Reply:
No, it's always enabled under normal circumstances. It's only set to false if you pass `--emit=llvm-ir`, or similar things that dump IR or assembly.